Code/Tools/ProjectManager/Platform/Mac/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    char enginePathParam[AZ_MAX_PATH_LEN];
+    char processPathStr[AZ_MAX_PATH_LEN];
+    uint32_t pathLen = AZ_MAX_PATH_LEN;
+    _NSGetExecutablePath(processPathStr, &pathLen);
+    snprintf(enginePathParam, AZ_MAX_PATH_LEN, "--engine-path=%s/../Engine", processPathStr);
+
+    char *commandLineParams[AZ_MAX_PATH_LEN] = {processPathStr, enginePathParam, nullptr};
+    
+    // Create a ComponentApplication to initialize the AZ::SystemAllocator and initialize the SettingsRegistry
+    AZ::ComponentApplication application(2, commandLineParams);
+    application.Create(AZ::ComponentApplication::Descriptor());
+
+    AZ::IO::FixedMaxPath processPath = AZ::IO::PathView(AZ::Utils::GetExecutableDirectory());
+    AZ::IO::FixedMaxPath enginePath = processPath/".."/"Engine";
```

Comment:
The executable directory doesn't require any setup of the ComponentApplication, so this code can be moved aboved

Reply:
Fixed